frontend/src/views/BoardGame.tsx

```diff
@@ -41,6 +42,14 @@ export const BoardGame: FC<BoardGameProps> = (props: BoardGameProps) => {
       setCurrentUsers(currentStatus.users);
     }
   }, [lastMessage]);
+  
+  useEffect(() => {
+    window.onbeforeunload = function () {
+      if (currentUsers.length || cards.length) {
```

Comment:
I believe `cards.length` will always be `>0`. In the `<Room />` component we check before rendering boardgame.

I can see in both cases, we could set the `when={true}` and it would be what we want.